kubernetes/structure_endpoint.go

```diff
@@ -0,0 +1,131 @@
+package kubernetes
+
+import (
+	api "k8s.io/api/core/v1"
+)
+
+func expandEndpointAddresses(in []interface{}) []api.EndpointAddress {
+	if len(in) == 0 {
+		return []api.EndpointAddress{}
+	}
+	addresses := make([]api.EndpointAddress, len(in))
+	for i, addr := range in {
+		r := api.EndpointAddress{}
+		addrCfg := addr.(map[string]interface{})
+
+		if v, ok := addrCfg["hostname"].(string); ok {
+			r.Hostname = v
+		}
+		if v, ok := addrCfg["ip"].(string); ok {
+			r.IP = v
+		}
+		if v, ok := addrCfg["node_name"].(string); ok {
+			if v != "" {
```

Comment:
you could pair this check with the `ok` about using `&&` and avoid the second `if`